### ISSUE TITLE
Multiple Simultaneous Ajax Requests (with one callback)

### ISSUE THREAD

**Issue by luckyshot** [pyrsmk/qwest#104]
Right now to run a callback after 2+ AJAX calls I am nesting them:

```
qwest.get( '/user/profile' ).then( function(xhr, profile ){
	qwest.get( '/user/books' ).then( function(xhr, books ){
		callback( profile, books );
	});
});
```

Yet this is unoptimal since I need to wait for one request to finish before the next one starts, and that's just with two requests. 
Is there a way in Qwest to take a list of promises and wait until all promises have resolved?

For example, in jQuery there's [$.when()](https://api.jquery.com/jquery.when/) and in Mithril there is [m.sync()](https://lhorie.github.io/mithril/mithril.sync.html).

**Comment by pyrsmk** [comment]
Maybe you should look for a [Promise.all()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) polyfill?

If you don't find anything lightweight/interesting, I would be glad to add the support to qwest ;)

**Comment by gagan-bansal** [comment]
+1

**Comment by pyrsmk** [comment]
I'll add this functionnality to qwest with : 

```js
qwest(['/user/profile', '/user/books' ]).then(function(args) {

});
```

**Comment by pyrsmk** [comment]
Of course, if we set `data` and `options` parameters, they will be set for the both requests.

**Comment by luckyshot** [comment]
That's great to hear @pyrsmk!

How about:

    qwest
     .method(url, data, options, before)
     .method(url, data, options, before)
     .then(function(xhr, responses) { });

**Comment by pyrsmk** [comment]
Good idea!

**Comment by pyrsmk** [comment]
But it's a bit complicated to implement since calling a method returns a promise.

**Comment by pyrsmk** [comment]
Included in the incoming `4.0` release
